extensions/http/src/main/java/org/mule/extension/http/api/HttpMessageBuilder.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.http.api;
+
+import static org.mule.runtime.core.config.i18n.MessageFactory.createStaticMessage;
+import org.mule.runtime.core.api.MuleRuntimeException;
+import org.mule.runtime.core.message.ds.ByteArrayDataSource;
+import org.mule.runtime.core.message.ds.StringDataSource;
+import org.mule.runtime.extension.api.annotation.Parameter;
+import org.mule.runtime.extension.api.annotation.param.Optional;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import javax.activation.DataHandler;
+import javax.activation.FileDataSource;
+
+/**
+ * Base component to create HTTP messages.
+ */
+public class HttpMessageBuilder
+{
+    @Parameter
+    @Optional
+    protected Map<String, String> headers = new HashMap<>();
+
+    //@Parameter
```

Comment:
This is commented to avoid issues until https://www.mulesoft.org/jira/browse/MULE-9713 is fixed.